client/src/pages/customer-pages/Home/index.js

```diff
@@ -55,39 +112,101 @@ class Home extends Component {
 
   handleCheck = e => {
```

Comment:
It looks like the flow here is that when the user checks "keep me updated" and then enters their phone and closes the modal, they have to click the "check" button again to proceed to their flight details.
It might be a better user experience to go right to the flight details when closing the modal, and then some of these long functions could be divided up a little bit.  Here's my idea:

1) Move the fetch to /api/v1/check_flight and all of that associated logic (sessionStorage, window.location, ..) into a new `this.gotoFlight` function.

2) `handleCheck` can then simplify to something like this:
```
if (checkboxValue === true) {
  this.openModal()
}
else {
  this.gotoFlight()
}
```

3) closeModal then does a network fetch and then, if successful, calls `this.gotoFlight()` right after setting modalIsOpen to false (or maybe even right before?)

My other suggestion is for the customer endpoints.  I don't see where there is an endpoint that is creating a customer model and saving the phone number.  `api/v1/check_flight` is looking for a customer with a given phone number, and `update_customer` is updating the notify_me flag for a customer matching the phone number you pass in - both of those expect that a customer already exists with that phone number.  The simplest data flow might be to have closeModal() be the spot where you do a post to create a customer (since you'll know the phone number, the flight number and the date), and then that might be the only endpoint you need relating to customers.  Maybe :D